plugins/catalog-react/report-alpha.api.md

```diff
@@ -324,6 +321,54 @@ export interface EntityContentLayoutProps {
   }>;
 }
 
+// @alpha (undocumented)
+export type EntityPredicate =
+  | EntityPredicateExpression
+  | EntityPredicatePrimitive
+  | {
+      $all: EntityPredicate[];
+    }
+  | {
+      $any: EntityPredicate[];
+    }
+  | {
+      $not: EntityPredicate;
+    };
+
+// @alpha (undocumented)
+export type EntityPredicateExpression = {
+  [KPath in string]: EntityPredicateValue;
+} & {
+  [KPath in `$${string}`]: never;
+};
+
+// @alpha (undocumented)
+export type EntityPredicatePrimitive = string | number | boolean;
```

Comment:
Hmm with the assumption that it gets stringified if made into a backend query later on, and that it somehow gets coerced or doesn't run into precision problems when running in-memory? I'm just thinking about what happens if you pass in `1/3` as the number, or if you check with equality with `4` but the entity contained `"4"`. Maybe we need to be very explicit here to not fall into a trap of implicit unclear behaviors 🤔

Reply:
Yep might be that we want all comparisons to be string based in the end, but on the other hand it's pretty likely we end up with some JSON-based filtering and then we have more strict types.

I think string vs number comparisons should probably always be handled though, so `4 = '4'` now.